usr/libexec/greenboot/greenboot.sh

```diff
@@ -2,75 +2,56 @@
 set -euo pipefail
 IFS=$'\n\t'
 
-run_required_scripts () {
-  echo "Running Required Health Check Scripts..."
-  local required_scripts=`find /etc/greenboot/check/required.d -name '*.sh'`
+script_runner () {
```

Comment:
One thing I usually do in bash functions is immediately assign the positional arguments to meaningful local vars. That makes it much easier to read. E.g.:

```bash
script_runner () {
  local scripts_dir=$1; shift
  local mode=$1; shift
  local message=$1; shift
  ...
}
```